### ISSUE TITLE
False positive "UnusedPrivateMember" on local val `in` (with apostrophes)

### ISSUE THREAD

**Issue by christianhujer** [detekt/detekt#3825]
## Expected Behavior
A local variable can be named `in` by using backticks.
This is for example interesting for code which is translated/converted from Java.
The name `in` is most likely not the issue of this bug.
The issue is most likely related to local variables being named with backticks in general.

## Observed Behavior
In the following code, the variable `in` was reported as `UnusedPrivateMember`:
```
    @Test
    fun testReadAllUBytes() {
        val `in` = ByteArrayInputStream(ubyteArrayOf("010203CAFEBABE")) as InputStream
        val expected = ubyteArrayOf("010203CAFEBABE")
        val actual = `in`.readAllUBytes()
        assertArrayEquals(expected, actual)
    }
```
Because `in` is clearly used, the expected behavior is that detekt does not report `UnusedPrivateMember` about `in`.
The actual behavior is that detekt does report `UnusedProvateMember` about `in`:
```
> Task :lib-nelkinda-core:detekt FAILED
style - 10min debt
        UnusedPrivateMember - [in] at /home/christian.hujer/git/nelkinda/xxx-xxx/lib-nelkinda-core/src/test/kotlin/com/nelkinda/java/io/InputStreamTest.kt:12:9
```

## Steps to Reproduce
Try the following source code:
```
    @Test
    fun testRead() {
        val `in` = StringReader("foo") as Reader
        val expected = "foo"
        val actual = `in`.readText()
        assertEquals(expected, actual)
    }
```

## Context
n/a

## Workaround
Use `@SuppressWarnings("UnusedPrviateMemger")` before the declaration of `in`.

## Your Environment

* Version of detekt used: 1.17.1
* Version of Gradle used (if applicable): 9.8.2
* Gradle scan link (add `--scan` option when running the gradle task): n/a
* Operating System and version: Kubuntu 20.04.2 LTS
* Link to your project (if it's a public repository): n/a

**Comment by cortinico** [comment]
> A local variable can be named `in` by using backticks.

True. The is actually a bit wider as:

```kotlin
val `hello` = 42
val hello = 42
```

Is not allowed in Kotlin, as ``hello`` and `hello` represents the same variable.
I've added an extra check to `UnusedPrivateMember` to make sure that we handle properly backticks: #3828
